browser/autocomplete/search_suggestions_query_check_constants.h

```diff
@@ -0,0 +1,1454 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_AUTOCOMPLETE_SEARCH_SUGGESTIONS_QUERY_CHECK_CONSTANTS_H_
+#define BRAVE_BROWSER_AUTOCOMPLETE_SEARCH_SUGGESTIONS_QUERY_CHECK_CONSTANTS_H_
+
+#include "base/containers/fixed_flat_map.h"
+
+inline constexpr double kProbHashThreshold = 0.015;
+
+inline constexpr auto kProbHashChars = base::MakeFixedFlatMap<char, int>(
+    {{'1', 1},  {'0', 0},  {'3', 3},  {'2', 2},  {'5', 5},  {'4', 4},
+     {'7', 7},  {'6', 6},  {'9', 9},  {'8', 8},  {'A', 36}, {'C', 38},
+     {'B', 37}, {'E', 40}, {'D', 39}, {'G', 42}, {'F', 41}, {'I', 44},
+     {'H', 43}, {'K', 46}, {'J', 45}, {'M', 48}, {'L', 47}, {'O', 50},
+     {'N', 49}, {'Q', 52}, {'P', 51}, {'S', 54}, {'R', 53}, {'U', 56},
+     {'T', 55}, {'W', 58}, {'V', 57}, {'Y', 60}, {'X', 59}, {'Z', 61},
+     {'a', 10}, {'c', 12}, {'b', 11}, {'e', 14}, {'d', 13}, {'g', 16},
+     {'f', 15}, {'i', 18}, {'h', 17}, {'k', 20}, {'j', 19}, {'m', 22},
+     {'l', 21}, {'o', 24}, {'n', 23}, {'q', 26}, {'p', 25}, {'s', 28},
+     {'r', 27}, {'u', 30}, {'t', 29}, {'w', 32}, {'v', 31}, {'y', 34},
+     {'x', 33}, {'z', 35}});
+
+inline constexpr double kProbHashLogM[496][62] = {
```

Comment:
What about putting into .pak as a binary?
Storing data in C++ headers doesn't look like reliable approach.o

Reply:
> Not sure how to comment as brave search's code is not opened.
It's originally from HumanWeb / WDP: https://github.com/brave/web-discovery-project/blob/a201c49963de25658228124ca5edb83b7474dc3d/modules/web-discovery-project/sources/web-discovery-project.es#L99